### PR TITLE
WIP Add GUI for ignoring scenes

### PR DESCRIPTION
**Pull request by b-camphart**
Moves the code for adding/ignoring scenes into a shared file (to avoid import cycles).  I considered using the context API, but it erases all the typing and, since all the logic used shared stores anyway, I figured this design couldn't hurt.

WIP because we may want to add a GUI for including scenes back into the project, but I'll wait to hear from you @kevboh :)

### CODE REVIEW

**Comment by b-camphart** [comment]
This should fix #173

File: src/view/explorer/UseCases.ts
Comment (kevboh) [inline src/view/explorer/UseCases.ts:1]: Could you rename this to follow existing file naming conventions? I would instead call this something like `scene-menu-items.ts`

File: src/view/explorer/ExplorerPane.ts [hunk 1]
```diff
@@ -203,6 +205,11 @@ export class ExplorerPane extends ItemView {
           item.setIcon("document");
           item.onClick(() => addRelativeScene("after", file));
         });
+        menu.addItem((item) => {
+          item.setTitle("Ignore scene");
```
Comment (kevboh) [inline src/view/explorer/ExplorerPane.ts:209]: Technically the file is not meant to be a scene, so I would instead call this

```suggestion
          item.setTitle("Ignore note in Longform");
```

File: src/view/explorer/ExplorerPane.ts [hunk 2]
```diff
@@ -203,6 +205,11 @@ export class ExplorerPane extends ItemView {
...
+          item.setIcon("document");
```
Comment (kevboh) [inline src/view/explorer/ExplorerPane.ts:210]:
```suggestion
          item.setIcon("minus-circle");
```

maybe?

File: src/view/explorer/ExplorerPane.ts [hunk 3]
```diff
@@ -203,6 +205,11 @@ export class ExplorerPane extends ItemView {
...
+          item.onClick(() => ignoreScene(file.name.split(".")[0]));
```
Comment (kevboh) [inline src/view/explorer/ExplorerPane.ts:211]: This breaks for files with multiple periods in the name—also non-markdown files won't appear here, so we can safely trim `.md` and use that instead.

File: src/view/explorer/SceneList.svelte
Comment (kevboh) [inline src/view/explorer/SceneList.svelte:1]: Please update this file to not include formatting changes—can't review like this

File: src/view/explorer/UseCases.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            const targetDraft = d[index] as MultipleSceneDraft;
+            (d[index] as MultipleSceneDraft).ignoredFiles = [
+                ...targetDraft.ignoredFiles,
+                ...targetDraft.unknownFiles,
+            ];
+            (d[index] as MultipleSceneDraft).unknownFiles = [];
+            return d;
+        });
+    }
+}
```
Comment (kevboh) [inline src/view/explorer/UseCases.ts:75]: files should always include final newlines

File: src/view/explorer/ExplorerPane.ts [hunk 5]
```diff
@@ -285,4 +292,4 @@ export class ExplorerPane extends ItemView {
       this.explorerView.$destroy();
     }
   }
-}
+}
```
Comment (kevboh) [inline src/view/explorer/ExplorerPane.ts:295]: Restore newline

**Review by kevboh — CHANGES_REQUESTED** [review]

**Comment by b-camphart** [comment]
The formatting thing was a nightmare 😅  But I think I resolved all the changes you asked for.  The question is, do you want to have me include a GUI for re-adding notes to the project?  Would probably require some new GUI design to do it.

**Review by kevboh — APPROVED** [review]

**Comment by kevboh** [comment]
@b-camphart no, this is fine.

**Pull request merged**